### ISSUE TITLE
sqlalchemy.exc.InvalidRequestError: Table 'langchain_pg_collection' is already defined for this MetaData instance. 

### ISSUE THREAD

**Issue by arezazadeh** [langchain-ai/langchain#14699]
### Issue you'd like to raise.

what is this issue, and how can i resolve it: 

```python
    os.environ["AZURE_OPENAI_API_KEY"] = AZURE_OPENAI_API_KEY
    os.environ["AZURE_OPENAI_ENDPOINT"] = AZURE_OPENAI_ENDPOINT
    os.environ["OPENAI_API_TYPE"] = "azure"
    os.environ["OPENAI_API_VERSION"] = "2023-05-15"
    os.environ["OPENAI_API_KEY"] = OPENAI_API_KEY

    embedding = OpenAIEmbeddings()

    COLLECTION_NAME = "network_team_documents"
    CONNECTION_STRING = PGVector.connection_string_from_db_params(
        driver=os.environ.get(DB_DRIVER, DB_DRIVER),
        host=os.environ.get(DB_HOST, DB_HOST),
        port=int(os.environ.get(DB_PORT, DB_PORT)),
        database=os.environ.get(DB_DB, DB_DB),
        user=os.environ.get(DB_USER, DB_USER),
        password=os.environ.get(DB_PASS, DB_PASS),
    )

    store = PGVector(
        collection_name=COLLECTION_NAME,
        connection_string=CONNECTION_STRING,
        embedding_function=embedding,
        extend_existing=True,
    )

    gpt4 = AzureChatOpenAI(
        azure_deployment="GPT4",
        openai_api_version="2023-05-15",
    )
    retriever = store.as_retriever(search_type="similarity", search_kwargs={"k": 10})

    qa_chain = RetrievalQA.from_chain_type(llm=gpt4,
                                           chain_type="stuff",
                                           retriever=retriever,
                                           return_source_documents=True)
    return qa_chain
```

```python
Traceback (most recent call last):
  File "/opt/network_tool/chatbot/views.py", line 21, in chat
    chat_object = create_session()
  File "/opt/network_tool/chatbot/chatbot_functions.py", line 95, in create_session
    store = PGVector(
  File "/opt/klevernet_venv/lib/python3.10/site-packages/langchain_community/vectorstores/pgvector.py", line 199, in __init__
    self.__post_init__()
  File "/opt/klevernet_venv/lib/python3.10/site-packages/langchain_community/vectorstores/pgvector.py", line 207, in __post_init__
    EmbeddingStore, CollectionStore = _get_embedding_collection_store()
  File "/opt/klevernet_venv/lib/python3.10/site-packages/langchain_community/vectorstores/pgvector.py", line 66, in _get_embedding_collection_store
    class CollectionStore(BaseModel):
  File "/opt/klevernet_venv/lib/python3.10/site-packages/sqlalchemy/orm/decl_api.py", line 195, in __init__
    _as_declarative(reg, cls, dict_)
  File "/opt/klevernet_venv/lib/python3.10/site-packages/sqlalchemy/orm/decl_base.py", line 247, in _as_declarative
    return _MapperConfig.setup_mapping(registry, cls, dict_, None, {})
  File "/opt/klevernet_venv/lib/python3.10/site-packages/sqlalchemy/orm/decl_base.py", line 328, in setup_mapping
    return _ClassScanMapperConfig(
  File "/opt/klevernet_venv/lib/python3.10/site-packages/sqlalchemy/orm/decl_base.py", line 578, in __init__
    self._setup_table(table)
  File "/opt/klevernet_venv/lib/python3.10/site-packages/sqlalchemy/orm/decl_base.py", line 1729, in _setup_table
    table_cls(
  File "", line 2, in __new__
  File "/opt/klevernet_venv/lib/python3.10/site-packages/sqlalchemy/util/deprecations.py", line 281, in warned
    return fn(*args, **kwargs)  # type: ignore[no-any-return]
  File "/opt/klevernet_venv/lib/python3.10/site-packages/sqlalchemy/sql/schema.py", line 436, in __new__
    return cls._new(*args, **kw)
  File "/opt/klevernet_venv/lib/python3.10/site-packages/sqlalchemy/sql/schema.py", line 468, in _new
    raise exc.InvalidRequestError(
sqlalchemy.exc.InvalidRequestError: Table 'langchain_pg_collection' is already defined for this MetaData instance.  Specify 'extend_existing=True' to redefine options and columns on an existing Table object.
```

**Comment by maiqingqiang** [comment]
langchain==0.0.348 is ok，the bug appears to have been caused by version 0.0.350 & 0.0.349.

#14463 #14414 @baskaryan @efriis 

https://github.com/langchain-ai/langchain/blob/6ceb8e2ad43856a23b07dcc5b2f780504e28d614/libs/langchain/langchain/vectorstores/pgvector.py#L104

https://github.com/langchain-ai/langchain/blob/dec277a637b7aeb645a0074431a2029fde5058f0/libs/community/langchain_community/vectorstores/pgvector.py#L63

**Comment by arezazadeh** [comment]
i had the issue on both 348 and 349, I was able to fix it by adding `__table_args__ = {'extend_existing': True}` to the LangChainTable and also downgraded my 349 to 348 and did the same thing and fixed the issue.

**Comment by ElReyZero** [comment]
Downgrading to 0.0348 worked for me without adding the extend_existing argument!

**Comment by efriis** [comment]
Thanks for the report folks! I'll look into this today

**Comment by efriis** [comment]
This should be fixed in the next release! Also, the others are correct that you shouldn't have the `extend_existing` param in your init

**Comment by nannarane** [comment]
> i had the issue on both 348 and 349, I was able to fix it by adding `__table_args__ = {'extend_existing': True}` to the LangChainTable and also downgraded my 349 to 348 and did the same thing and fixed the issue.

hello. I also had the same problem,
__table_args__ = {'extend_existing': True}

Can you tell me which part of the code you directly inserted?
Can you tell me how you handled it without putting it directly in that pip package?

**Comment by efriis** [comment]
#14726

**Comment by arezazadeh** [comment]
I added it in this file: 
`/lib/python3.10/site-packages/langchain/vectorstores/_pgvector_data_models.py ` 

```
class CollectionStore(BaseModel):
    """Collection store."""

    __tablename__ = "langchain_pg_collection"
    __table_args__ = {'extend_existing': True}
```

**Comment by reneric** [comment]
Why is `langchain_pg_collection` being used exactly? Is that something happening under the hood when attempting to initialize another collection?

**Comment by ElReyZero** [comment]
I tried the fix after updating to version 0.0.351 but the bug is still present.

**Comment by maiqingqiang** [comment]
0.0.351

raise exc.InvalidRequestError(
sqlalchemy.exc.InvalidRequestError: Multiple classes found for path "EmbeddingStore" in the registry of this declarative base. Please use a fully module-qualified path.

**Comment by maiqingqiang** [comment]
> 0.0.351
> 
> raise exc.InvalidRequestError( sqlalchemy.exc.InvalidRequestError: Multiple classes found for path "EmbeddingStore" in the registry of this declarative base. Please use a fully module-qualified path.

Why utilize the _get_embedding_collection_store method instead of following the approach used in version 0.0.348?
@efriis @baskaryan

**Comment by nannarane** [comment]
For inquiries about https://github.com/langchain-ai/langchain/issues/14699#issuecomment-1857393076, we registered this issue because it was installed through pip when building the docker image to run on Kubernetes.

I solved it by lowering the version to 0.0.311. It seems that https://github.com/langchain-ai/langchain/commit/c81d0f9290c73b95cdae27eb386f562285e24c9c has been treated as second-rate now.
